05/DB.test.js

```diff
@@ -1,83 +1,105 @@
-         expect.assertions(1)
-         const id = 123;
-         try {
-            await db.select(id)
-         }
-         catch (e) {
-            expect(e).toBe('ID not found')
-         }
-      })
-  })
-  describe('remove', async () => {
-    it('reject to remove not existing id', async () => {
+      try {
+        await db.insert({ id: 1, name: "Kevin" });
+      } catch (e) {
+        expect(e).toBe("ID can't be duplicated!");
+      }
+    });
+    it("insert data to DB and generates id if not provided", async () => {
+      const data = { name: "Joe" };
+      const result = await db.insert(data);
+      expect(result).toHaveProperty("id");
+      expect(result.name).toEqual(data.name);
+    });
+  });
+  describe("select", async () => {
+    it("select data by id", async () => {
+      const result = await db.insert({ id: 1, name: "John" });
+      const selectData = await db.select(1);
+      expect(selectData).toEqual(result);
+    });
+    it("show error when id not exist", async () => {
+      expect.assertions(1);
       const id = 123;
       try {
-         await db.remove(id)
+        await db.select(id);
+      } catch (e) {
+        expect(e).toBe("ID not found");
       }
-      catch (e) {
-         expect(e).toBe('Item not exist!')
+    });
+  });
+  describe("remove", async () => {
+    it("reject to remove not existing id", async () => {
+      expect.assertions(1);
+      const id = 123;
+      try {
+        await db.remove(id);
+      } catch (e) {
+        expect(e).toBe("Item not exist!");
       }
-    })
-    it('remove item from db when id exist', async () => {
-      const data = {name: 'Joe', id: 1};
+    });
+    it("remove item from db when id exist", async () => {
+      const data = { name: "Joe", id: 1 };
       await db.insert(data);
       await db.remove(1);
-    })
-  })
-//   describe('update', async () => {
-//      it('reject update data without id', async () => {
-      
-//      })
-//   })
-})
+    });
+  });
+  describe("update", async () => {
+    it("reject update data without id", async () => {
+      expect.assertions(1);
+      const data = { name: "John" };
+      try {
+        await db.update(data);
+      } catch (e) {
+        expect(e).toBe("ID have to be set!");
+      }
+    });
+    it("update data when id exist", async () => {
+      const data = { name: "Adam", id: 2 };
+      await expect(db.update({ name: "Kate", id: data.id }));
+    });
+  });
+ 
+   it("clears all rows in the database", async () => {
+      await db.insert({ id: 1, name: "John" });
+      await db.insert({ id: 2, name: "Joe" });
+      await db.truncate();
+      const rows = await db.getRows();
+      expect(rows.length).toBe(0);
+     });
 
+   it('get all rows in the database', async () => {
+      await db.insert({ id: 1, name: "John" });
+      await db.insert({ id: 2, name: "Joe" });
+      const rows = await db.getRows()
+
+      expect(rows.length).toBe(2)
+   })
```

Comment:
Testy wyglądają ok - wartość sprawdzić każdy z osobna (np. przez dodanie `.only` tj. `it.only('get all rows in the database', async () => {`) i zobaczyć czy przechodzą. Jeśli tak to oznacza, że one są ok, ale inny test wpływa na negatywny wynik.

Okazało się, że w linii 85 masz test, który nic nie sprawdza (brakuje `.toBe` czy `.toThrow`) i to ma negatywny wpływ na następne testy :)